src/components/ModalOverlay/ModalOverlay.jsx

```diff
@@ -0,0 +1,38 @@
+
+import React, { forwardRef, useImperativeHandle, useRef } from "react";
+import PropTypes from 'prop-types';
+import styles from './ModalOverlay.module.css';
+
+const ModalOverlay = forwardRef(({ handlerClose, children }, ref) => {
+
+    const refContainer = useRef();
+
+    const closeSmooth = () => {
+        refContainer.current.classList.remove(styles.opened);
+        setTimeout(handlerClose, 400);
+    };
+
+    useImperativeHandle(ref, ()=> ({
+        closeSmooth: ()=> closeSmooth()
```

Comment:
Можно лучше:
так как тут функция `closeSmooth` вызывается без аргументов, и ее имя соответствует значению ключа, можно передать просто 
```suggestion
    useImperativeHandle(ref, ()=> ({
        closeSmooth
```